riptide-problem/src/main/java/org/zalando/riptide/problem/ProblemRoute.java

```diff
@@ -10,17 +14,51 @@
 
 public final class ProblemRoute {
 
-    private static final Route ROUTE = dispatch(contentType(),
-            on(parseMediaType("application/problem+json")).call(Exceptional.class, Exceptional::propagate),
-            on(parseMediaType("application/x.problem+json")).call(Exceptional.class, Exceptional::propagate),
-            on(parseMediaType("application/x-problem+json")).call(Exceptional.class, Exceptional::propagate));
+    private static final MediaType PROBLEM = parseMediaType("application/problem+json");
+
+    /**
+     * @see <a href="http://zalando.github.io/restful-api-guidelines/common-data-objects/CommonDataObjects.html#must-use-problem-json">Zalando RESTful API Guidelines</a>
+     */
+    private static final MediaType X_DOT_PROBLEM = parseMediaType("application/x.problem+json");
+
+    /**
+     * Alternative spelling for {@link #X_DOT_PROBLEM}.
+     */
+    private static final MediaType X_DASH_PROBLEM = parseMediaType("application/x-problem+json");
+
+    private static final Route PROPAGATE = problemHandling(Exceptional::propagate);
 
     ProblemRoute() {
         // package private so we can trick code coverage
     }
 
+    /**
+     * Produces a {@link Route route} that dispatches on the {@link Navigators#contentType() content type} and
+     * recognises {@code application/problem+json} as well as {@code application/x-problem+json} and
+     * {@code application/x.problem+json} as {@link Problem problems} and {@link Exceptional#propagate() propagates}
+     * them.
+     *
+     * @see #problemHandling(ThrowingConsumer)
+     * @see Exceptional#propagate()
+     * @return static route for handling problems by propagating them as exceptions
+     */
     public static Route problemHandling() {
-        return ROUTE;
+        return PROPAGATE;
+    }
+
+    /**
+     * Produces a {@link Route route} that dispatches on the {@link Navigators#contentType() content type} and
+     * recognises {@code application/problem+json} as well as {@code application/x-problem+json} and
+     * {@code application/x.problem+json} as {@link Problem problems} and handles them given the supplied consumer.
+     *
+     * @param consumer the exception handler
+     * @return a route for handling problems dynamically
+     */
+    public static Route problemHandling(final ThrowingConsumer<Exceptional, ? extends Exception> consumer) {
+        return dispatch(contentType(),
+                on(PROBLEM).call(Exceptional.class, consumer),
```

Comment:
I prefer individual bindings per value. I changed the implementation to show how to deal with a situation like that. It's a bit more verbose, but I think it's also more readable/understandable.